File: wallet/src/main/java/com/worldventures/wallet/ui/settings/general/profile/WalletSettingsProfilePresenter.java
package com.worldventures.wallet.ui.settings.general.profile;

import com.worldventures.core.modules.picker.model.PhotoPickerModel;
import com.worldventures.wallet.ui.common.base.WalletPresenter;

public interface WalletSettingsProfilePresenter extends WalletPresenter<WalletSettingsProfileScreen> {

   void openDisplaySettings();

   void handleDoneAction();

   boolean isDataChanged();

   void handlePickedPhoto(PhotoPickerModel photoPickerModel);

   void choosePhoto();

   void doNotAdd();

   void goBack();

   void confirmDisplayTypeChange();

   void handleBackOnDataChanged();

   void revertChanges();
}

File: wallet/src/main/java/com/worldventures/wallet/ui/settings/general/profile/WalletSettingsProfilePresenter.kt
package com.worldventures.wallet.ui.settings.general.profile

import com.worldventures.core.modules.picker.model.PhotoPickerModel
import com.worldventures.wallet.ui.common.base.WalletPresenter

interface WalletSettingsProfilePresenter : WalletPresenter<WalletSettingsProfileScreen> {

   fun openDisplaySettings()

   fun handleDoneAction()

   fun handlePickedPhoto(photoPickerModel: PhotoPickerModel)

   fun choosePhoto()

   fun doNotAdd()

   fun goBack()

   fun confirmDisplayTypeChange()

   fun handleBackOnDataChanged()

   fun revertChanges()
}
